components/salsa-macros/src/update.rs

```diff
@@ -52,14 +52,18 @@ pub(crate) fn update_derive(input: syn::DeriveInput) -> syn::Result<TokenStream>
                     let field_ty = &binding.ast().ty;
                     let field_index = Literal::usize_unsuffixed(index);
 
+                    let update_field = quote_spanned! {
+                        binding.binding.span() =>
+                        unsafe {
+                            salsa::plumbing::UpdateDispatch::<#field_ty>::maybe_update(
+                                #binding,
+                                #new_value.#field_index,
+                            )
+                        }
+                    };
```

Comment:
I hoped that this would give us better compiler errors that point to the relevant field but it seems that using spanned here doesn't help